authorize/evaluator/functions.go

```diff
@@ -41,13 +41,14 @@ func isValidClientCertificate(ca, cert string) (bool, error) {
 	}
 
 	_, verifyErr := xcert.Verify(x509.VerifyOptions{
-		Roots: roots,
+		Roots:     roots,
+		KeyUsages: []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
 	})
 	valid := verifyErr == nil
 
 	isValidClientCertificateCache.Add(cacheKey, valid)
 
-	return valid, nil
+	return valid, verifyErr
```

Comment:
Is there a gotcha by having this error set here? \cc @calebdoxsey

Reply:
Yes. Error means something unexpected happened. If a user supplies an invalid cert its not an error, but it returns false.

Reply:
👍  changed to a log message in debug